### PR TITLE
W-16300511 Adding more information on types in json moduleloader and gaps with f…

### PR DESCRIPTION
**Pull request by martincousido**
…ormats and restrictions

# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

**Review by martincousido — COMMENTED** [review]
LGTM

File: modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc [hunk 1]
```diff
@@ -257,4 +257,150 @@ else -> "NO TYPE MATCHED"
...
+| `boolean`| `Boolean`
+| `array`| `Array`
+| `enums`| `Union`
+| `null`| `Null`
+| `object`| `Object`
+|===
+ 
+=== Limitations
+
+* Formats on JSON schema string types aren't supported and are transformed to `String` without any validation or restriction.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc]:
```suggestion
* DataWeave doesn't support formats on JSON schema string types and transforms these types to `String` without any validation or restriction.
```

File: modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc [hunk 2]
```diff
@@ -257,4 +257,150 @@ else -> "NO TYPE MATCHED"
...
+* Restrictions on number types such as maximums or minimums aren't supported and are be transformed to `Number` without any validation or restriction.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc]:
```suggestion
* DataWeave doesn't support restrictions on number types such as maximums or minimums and transforms these types to `Number` without any validation or restriction.
```

File: modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc [hunk 3]
```diff
@@ -257,4 +257,150 @@ else -> "NO TYPE MATCHED"
...
+* Patterns aren't supported.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc]:
```suggestion
* DataWeave doesn't support patterns.
```

File: modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc [hunk 4]
```diff
@@ -257,4 +257,150 @@ else -> "NO TYPE MATCHED"
...
+
+=== Object types
+
+Non-required properties in JSON Schema object types are marked as optional properties in the DataWeave Object type.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc]:
```suggestion
DataWeave marks non-required properties in JSON Schema object types as optional properties in the DataWeave Object type. 
```

**Review by Cristian-Venticinque — DISMISSED** [review]
Some suggestions to avoid passive voice

**Review by Cristian-Venticinque — APPROVED** [review]

**Pull request merged**